pwnlib/elf/elf.py

```diff
@@ -841,6 +841,13 @@ def _patch_elf_and_read_maps(self):
             log.warn_once("Injected /proc/self/maps code did not execute correctly")
             return {}
 
+        # Sometimes the original binary already fail to run, for example, in case glibc mismatches.
+        try:
+            int(data.split('-', 1)[0])
```

Comment:
```suggestion
            int(data.split('-', 1)[0], 16)
```

Need base16 as the addresses are hexadecimal. Have you tested this change locally? :wink:

Reply:
Nice spot! I'm sorry I missed this! I have 2 copy of this patch: one in venv and one in the fork. They were out of sync. Fixed and tested on my venv!